mdagent/tools/base_tools/analysis_tools/secondary_structure.py

```diff
@@ -0,0 +1,301 @@
+    np.save(file_path, values)
+
+    path_registry.map_path(
+        file_id,
+        file_name,
+        description=f"{x} values for trajectory with id: {traj_id}",
+    )
+    return file_id
+
+
+class ComputeDSSP(BaseTool):
+    name = "ComputeDSSP"
+    description = """Compute the DSSP (secondary structure) assignment
+    for a protein trajectory. Input is a trajectory file (e.g., .xtc, .
+    trr) and an optional topology file (e.g., .pdb, .prmtop). The output
+    is an array with the DSSP code for each residue at each time point."""
+    path_registry: PathRegistry | None = None
+    simplified: bool = True
+
+    def __init__(self, path_registry, simplified: bool = True):
+        super().__init__()
+        self.path_registry = path_registry
+        self.simplified = simplified
+
+    def _dssp_codes(self):
+        if self.simplified:
+            return ["H", "E", "C"]
+        return ["H", "B", "E", "G", "I", "T", "S", " "]
+
+    def _dssp_natural_language(self):
+        if self.simplified:
+            return {"H": "helix", "E": "strand", "C": "coil"}
+        return {
+            "H": "alpha helix",
+            "B": "beta bridge",
+            "E": "extended strand",
+            "G": "three helix",
+            "I": "five helix",
+            "T": "hydrogen bonded turn",
+            "S": "bend",
+            " ": "loop or irregular",
+        }
+
+    def _convert_dssp_counts(self, dssp_counts: dict):
+        code_to_description = self._dssp_natural_language()
+
+        descriptive_counts = {
+            code_to_description[code]: count for code, count in dssp_counts.items()
+        }
+        return descriptive_counts
+
+    def _summarize_dssp(self, dssp_array):
+        dssp_codes = self._dssp_codes()
+        # turn into dict where keys are codes
+        dssp_dict = {code: 0 for code in dssp_codes}
+        for frame in dssp_array:
+            for code in frame:
+                dssp_dict[code] += 1
+        return self._convert_dssp_counts(dssp_dict)
+
+    def _compute_dssp(self, traj):
+        return md.compute_dssp(traj, simplified=self.simplified)
+
+    def _run(self, traj_file, top_file=None):
+        try:
+            traj = load_single_traj(
+                path_registry=self.path_registry,
+                traj_fileid=traj_file,
+                top_fileid=top_file,
+            )
+            if not traj:
+                raise Exception("Trajectory could not be loaded.")
+        except Exception as e:
+            return str(e)
+
+        dssp_array = self._compute_dssp(traj)
+        write_raw_x("dssp", dssp_array, traj_file, self.path_registry)
+        summary = self._summarize_dssp(dssp_array)
+        return summary
+
+    async def _arun(self, traj_file, top_file=None):
+        raise NotImplementedError("Async version not implemented")
+
+
+class ComputeGyrationTensor(BaseTool):
+    name = "ComputeGyrationTensor"
+    description = """Compute the gyration tensor for each frame in a
```

Comment:
In the other tool descriptions I try to include trajectory file ID , instead of traj file (.xtc, .trr) as an old issue was the agent using the full path to get the file, instead of the file ID.